### PR TITLE
docs: add PR template reminder, Makefile, and assets folder

### PR DESCRIPTION
**Pull request by carlydf**
## What changed?
- Add PR template reminder to update docs
- Add first d2 diagram to Matching README (more to follow)
- Refactor `docs` folder to contain an `assets` folder where we store images for all the READMEs in the project, instead of using github assets

## Why?
To improve how we interact with and update docs

## How did you test it?
n/a

## Potential risks
n/a

## Is hotfix candidate?
no

### CODE REVIEW

File: service/history/README.md [hunk 1]
```diff
@@ -1 +1,287 @@
-See [docs/history-service.md](../../docs/history-service.md).
+# History Service
+
+This page is an introduction to the History Service architecture and internal implementation.
+See [Temporal overview](../../docs/README.md) for a system architecture overview placing this in context.
+
+<!-- https://lucid.app/lucidchart/0202e4b8-5258-4cd6-a6a0-67159300532b/edit -->
+<img src="https://github.com/temporalio/temporal/assets/52205/47a7c139-1bb4-46f9-a401-793a63e39341">
```
[the remark this answers is not in the document]
Comment (stephanos) [reply service/history/README.md:7]: Agreed! Will do that now.

**Comment by stephanos** [comment]
Note for reviewers: I believe [d2](https://d2lang.com/) is a useful diagram tool that has some notable advantages over Mermaid: it looks gorgeous and it has more options to get the layout you want (specifically, grid layouts). The downside is that it requires an extra rendering step since it's not natively supported like Mermaid. But in my experience, the diagrams you can create in d2 wouldn't even work in Mermaid - you'd have to use external tools.

File: docs/assets/matching-context.d2 [hunk 2]
```diff
@@ -0,0 +1,61 @@
+# https://d2lang.com
```
Comment (dandavison) [inline docs/assets/matching-context.d2:1]: Not important but we might decide at some point that we'd like to use separate directories for the source (d2) vs the build artifacts (SVGs).

**Review by dandavison — APPROVED** [review]
LGTM!

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -10,5 +10,9 @@
 ## Potential risks
 <!-- Assuming the worst case, what can be broken when deploying this change to production? -->
 
+## Documentation
+<!-- Have you made sure this change doesn't falsify anything currently stated in `docs/` If significant
+new behavior is added, have you described that in `docs/`? -->
```
Comment (dandavison) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
<!-- Have you made sure this change doesn't falsify anything currently stated in `docs/`? If significant
new behavior is added, have you described that in `docs/`? -->
```

**Review by stephanos — APPROVED** [review]

**Pull request merged**